File: regexlib/javascript_test_file/regexlib_1707.js
/* 1707
 * (<log4j:Event logger=")(.*?)(" timestamp=")(.*?)(" level=")(.*?)(" thread=")(.*?)(">)(.*?)(<log4j:Message><!\[CDATA\[)(.*?)(\]\]>)
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:""+"<log4j:Event logger="" timestamp="" level="" thread=""*64+"! _1SLQ_2"
 */
var REGEX = /(<log4j:Event logger=")(.*?)(" timestamp=")(.*?)(" level=")(.*?)(" thread=")(.*?)(">)(.*?)(<log4j:Message><!\[CDATA\[)(.*?)(\]\]>)/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '<log4j:Event logger=\"\" timestamp=\"\" level=\"\" thread=\"'.repeat(i*1) + '! _1SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}

File: regexlib/python_re_test_file/regexlib_1707.py
# 1707
# (<log4j:Event logger=")(.*?)(" timestamp=")(.*?)(" level=")(.*?)(" thread=")(.*?)(">)(.*?)(<log4j:Message><!\[CDATA\[)(.*?)(\]\]>)
# EXPONENT
# nums:4
# EXPONENT AttackString:""+"<log4j:Event logger="" timestamp="" level="" thread=""*64+"! _1SLQ_2"

import re
from time import perf_counter

regex = """(<log4j:Event logger=")(.*?)(" timestamp=")(.*?)(" level=")(.*?)(" thread=")(.*?)(">)(.*?)(<log4j:Message><!\[CDATA\[)(.*?)(\]\]>)"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "<log4j:Event logger=\"\" timestamp=\"\" level=\"\" thread=\"" * i * 1 + "! _1SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")